---
title: "Lab 10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Lab 10: Population genomics based on high throughput sequencing (HTS)

```{r}
library(vcfR)

```

```{r}
data(vcfR_example)
vcf

```

```{r}
strwrap(vcf@meta[1:7])
```

```{r}
queryMETA(vcf)
```

```{r}
queryMETA(vcf, element = 'DP')
```

```{r}
queryMETA(vcf, element = 'FORMAT=<ID=DP')
```

```{r}
head(getFIX(vcf))
```

```{r}
vcf@gt[1:6, 1:4]
```

```{r, eval = FALSE}
vcf <- read.vcfR("Grunwald/pinfsc50_filtered.vcf.gz")
```
```{r}
head(vcf)
```
```{r}
write.vcf(vcf, "myVCFdata_filtered.vcf.gz")
```

### Exercises Pt 1

# 1
```{r}
?read.vcfR
```

# 2
```{r}
queryMETA(vcf, element = 'AD')
```

# 3
```{r}
tail(vcf@fix)
```

# 4
```{r}
plot(vcf)
```
```{r}
library(ggplot2)
qplot(getQUAL(vcf), geom = "histogram")
```
# 5
```{r}
colnames(vcf@gt)
```

### Part II - Analysis of Genome Data


```{r}
library('vcfR')
vcf <- read.vcfR("Grunwald/pinfsc50_filtered.vcf.gz")
# This data can also be download from the course repo - 
https://github.com/jeffreyblanchard/EvoGeno/blob/master/Grunwald/pinfsc50_filtered.vcf.gz
```

```{r}
vcf
```

```{r}
x <- vcfR2genlight(vcf)

```

```{r}
x
```

```{r}
gt <- extract.gt(vcf, element = "GT")
gt[c(2,6,18), 1:3]
```

```{r}
t(as.matrix(x))[c(1,5,17), 1:3]
```

```{r}
library(adegenet)
```
```{r}
pop(x) <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
popNames(x)
```

```{r}
ploidy(x) <- 2
```

## Distance Matrices

```{r}
x.dist <- dist(x)
```

```{r}
x.dist <- poppr::bitwise.dist(x)
```


# ChromR Objects